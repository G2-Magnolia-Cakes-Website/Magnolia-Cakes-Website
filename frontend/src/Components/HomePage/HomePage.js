import React from "react";

const HomePage = () => {
  return <div>Magnolia Cakes</div>;
};

export default HomePage;
