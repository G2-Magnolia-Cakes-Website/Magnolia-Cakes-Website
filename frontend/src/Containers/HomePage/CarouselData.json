[
  { "url": "http://localhost:3000/carousel/carousel-birthday.jpg", "title": "birthday" },
  { "url": "http://localhost:3000/image-2.jpg", "title": "boat" },
  { "url": "http://localhost:3000/image-3.jpg", "title": "forest" },
  { "url": "http://localhost:3000/image-4.jpg", "title": "city" },
  { "url": "http://localhost:3000/image-5.jpg", "title": "italy" }
]
