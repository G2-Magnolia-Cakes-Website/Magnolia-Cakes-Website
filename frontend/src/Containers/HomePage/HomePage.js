import React from "react";
import AboutUsSection from "./Sections/AboutUsSection";
import SubheadingDivider from "../../Components/SubheadingDivider/SubheadingDivider";
import CarouselGallery from "../../Components/Carousel/CarouselGallery";
import LocationPage from "../LocationPage/LocationPage";

const HomePage = () => {
  return (
    <>
      <CarouselGallery />
      {/* <SubheadingDivider subheadingText="Cakes for all occasions" /> */}
      {/* <SubheadingDivider subheadingText="About Us" /> */}
      <AboutUsSection />
      <LocationPage />
    </>
  );
};

export default HomePage;
