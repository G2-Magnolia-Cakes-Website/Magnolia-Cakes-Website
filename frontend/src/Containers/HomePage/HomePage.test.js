import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import HomePage from "./HomePage";

jest.mock("swiper/react", () => ({
  Swiper: ({ children }) => children,
  SwiperSlide: ({ children }) => children,
}));

jest.mock("swiper/modules", () => ({
  Autoplay: (props) => null,
  EffectFade: (props) => null,
  Navigation: (props) => null,
  Pagination: (props) => null,
  Keyboard: (props) => null,
}));

jest.mock("swiper/css", () => jest.fn());

describe("test HomePage", () => {
  it("renders HomePage", () => {
    const wrapper = renderer
      .create(
        <BrowserRouter>
          <HomePage />
        </BrowserRouter>
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
