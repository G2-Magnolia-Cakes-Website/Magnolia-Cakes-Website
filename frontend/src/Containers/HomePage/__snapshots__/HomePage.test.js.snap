// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test HomePage renders HomePage 1`] = `
Array [
  <div>
    <div
      className="slide-wrapper"
    >
      <img
        alt="Birthday"
        src="carousel/carousel-birthday.jpg"
      />
    </div>
    <div
      className="slide-wrapper"
    >
      <img
        alt="Wedding & Anniversary"
        src="carousel/carousel-wedding-ann.jpg"
      />
    </div>
  </div>,
  <div
    className="about-us-section"
  >
    <img
      alt="About Us Portrait"
      className="about-us-portrait"
      src="about-us-portrait.jpg"
    />
    <div
      className="about-us-text"
    >
      <h2>
        About Us
      </h2>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. In at rutrum massa. Aliquam faucibus id lacus nec hendrerit. Nunc ac justo eget nibh lobortis pellentesque id eget sapien. In id placerat dui. Vestibulum sollicitudin turpis magna, vel convallis mi dictum in. Pellentesque consequat suscipit velit. Pellentesque vestibulum dui tellus, non vehicula erat efficitur nec. Nunc ullamcorper eu metus a sodales. Integer vitae bibendum ligula.
      </p>
      <button
        className="square-button unfilled"
        onClick={[Function]}
      >
        Read More
      </button>
    </div>
  </div>,
  <div
    className="location-page-wrapper"
  >
    <div
      className="location-info"
    >
      <h2>
        Where to find us?
      </h2>
      <p>
        Melbourne
        <br />
        VIC 3752
      </p>
      <h2>
        Business Hours
      </h2>
      <p>
        Monday - Sunday
        <br />
        9:00 am - 18:00 pm
      </p>
    </div>
  </div>,
]
`;
